client/src/templates/Challenges/components/fill-in-the-blanks.tsx

```diff
@@ -33,8 +34,63 @@ const AnswerText = ({ answer }: { answer: string }) => {
   );
 };
 
+type BlankInputProps = {
+  blankIndex: number;
+  answer: string;
+  isCorrect: boolean | null;
+  className: string;
+  onChange: (index: number, value: string) => void;
+  ariaLabel: string;
+  inputType?: 'pinyin-to-hanzi' | 'pinyin-tone';
+};
+
+const BlankInput = ({
```

Comment:
This component handles three inputs:
- Default input for non-Chinese
- Pinyin-to-Hanzi input (added in this PR)
- Pinyin Tone input (#64085)

The Pinyin Tone input doesn't involve Hanzi. The answer values are normal strings like ma, má, mǎ, etc.